KsApi/models/graphql/adapters/Reward+RewardFragment.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+
+extension Reward {
+  /**
+   Creates a `Reward` from a `RewardFragment`.
+
+    - parameter reward: The `RewardFragment` data structure.
+    - parameter projectId: The associated Project's ID'.
+    - parameter dateFormatter: A DateFormatter configured with the format "yyyy-MM-DD".
+    - parameter expandedShippingRules: Expanded shipping rules to be included.
+
+    - returns: A Reward.
+   */
+
+  static func reward(
+    from rewardFragment: GraphAPI.RewardFragment,
+    projectId: Int,
+    dateFormatter: DateFormatter = DateFormatter.isoDateFormatter,
+    expandedShippingRules: [ShippingRule]? = nil
```

Comment:
I guess the `RewardFragment` doesn't support the `expandedShippingRules` property on GQL yet? Is this also something that only comes from v1?

Reply:
Not 100 percent sure about this but I think expandedShippingRules only come back from the GraphQL call for addOns and that's why they have to be optional here?

Reply:
Yeah so I'm not adding it to the fragment because it's only needed in the context of add-ons selection. Instead we'll pass it into the function here from the encompassing query to not fetch this unnecessarily, I believe it was an expensive operation to fetch the expanded rules.